pgosupport.cmake

```diff
@@ -37,6 +37,18 @@ function(add_pgo TargetName)
     endforeach(ConfigType)
 endfunction(add_pgo)
 
+set(CLR_CMAKE_OPTDATA_PACKAGEID "optimization.PGO.CoreCLR")
+set(CLR_CMAKE_OPTDATA_PACKAGEWITHRID "optimization.${CLR_CMAKE_TARGET_OS}-${CLR_CMAKE_TARGET_ARCH}.PGO.CoreCLR")
+
+# Parse optdata package version from project.json
+file(TO_NATIVE_PATH "${CMAKE_SOURCE_DIR}/extract-from-json.py" ExtractFromJsonScript)
+file(TO_NATIVE_PATH "${CMAKE_SOURCE_DIR}/src/.nuget/optdata/project.json" OptDataProjectJsonPath)
+execute_process(
```

Comment:
I see, I have not realized that the result forms a compiler option. Then it seems that if you ran the script out cmake before cmake is executed and passed the resulting path to cmake in a variable (using the -DVARIABLE_NAME="XXXX" command line option, adding it to the __ExtraCmakeArgs), it would work well. CMake can detect that it was passed a different value of the variable and rebuild the VS / make projects in that case. That would mean that any time the config.json changes, the change would be picked up even without cleaning the bin folder.